tests/bugs/specialize-existential-in-generic.slang

```diff
@@ -20,7 +20,7 @@ struct Impl : IFoo
 
 struct GenType<T : IFoo>
 {
-    T obj;
+    T obj = T();
```

Comment:
Does this call the constructor of IFoo? Or the constructor of the actual type that implements IFoo?

If it is the first case, I am not sure if it will be valid to call the constructor of an interface for initialization.

If it is the second case and the actual type has an explicit constructor, should we still need to use `= T();` explicitly?